src/bandit-evaluator.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  BanditCategoricalAttributeCoefficients,
+  BanditModelData,
+  BanditNumericAttributeCoefficients,
+} from './interfaces';
+import { Attributes } from './types';
+
+export interface BanditEvaluation {
+  flagKey: string;
+  subjectKey: string;
+  subjectAttributes: Attributes;
+  actionKey: string;
+  actionAttributes: Attributes;
+  actionScore: number;
+  actionWeight: number;
+  gamma: number;
+  optimalityGap: number;
+}
+
+export class BanditEvaluator {
+  public evaluateBandit(
```

Comment:
This method will be tested and fleshed out next PR.
In the meantime, this PR covers the first few private methods it employs.